_api-reference/document-apis/get-documents.md

```diff
@@ -48,6 +47,79 @@ _source_includes | String | A comma-separated list of source fields to include i
+
+The Get Document API in OpenSearch operates in realtime mode by default, meaning it retrieves the latest version of the document regardless of the index's refresh rate (which determines when new data becomes searchable). However, if you request stored fields (using the `stored_fields` parameter) for a document that has been updated but not yet refreshed, the Get Document API needs to parse and analyze the document's source to extract those stored fields. To disable this realtime behavior and retrieve the document based on the last refreshed state of the index, set the `realtime` parameter to false.
+
+### Source filtering
+
+By default, the Get Document API returns the entire contents of the `_source` field for the requested document. However, you can choose to exclude the `_source` field from the response by using the `_source` URL parameter and setting it to false, as shown in the following example:
+
+```json
+GET test-index/_doc/0?_source=false
+```
+
+#### `source` includes and excludes
+
+If you only need specific fields from the `_source`, use the `_source_includes` or `_source_excludes` parameters to include or exclude particular fields, respectively. This can be beneficial for large documents, as retrieving only the required fields can reduce network overhead. Both parameters accept a comma-separated list of fields or wildcard expressions, as shown in the following example where any `_source` that contains `*.play` is included in the response, but excludes sources with the field `entities`:
+
+```
+GET test-index/_doc/0?_source_includes=*.play&_source_excludes=entities
+```
+
+#### Shorter notation
+
+If only want to include certain fields and don't need to exclude any, you can use a shorter notation by specifying the desired fields directly in the `_source` parameter:
+
+```json
+GET test-index/_doc/0?_source=*.id
+```
+
+### Routing
+
+When indexing documents in OpenSearch, specify a `routing` value to control the shard assignment for those documents. If routing was used during indexing, you must provide the same routing value when retrieving the document using the Get Document API, as shown in the following example:
+
+```json
+GET test-index/_doc/1?routing=user1
+```
+
+This request retrieves the document with the ID `1`, but it uses the routing value "user1" to determine the shard where the document is stored. If the correct routing value is not specified, the Get Document API will not be able to locate and fetch the requested document.
+
+### Preference
+
+The Get Document API allows you to control which shard replica should handle the request. By default, the operation is randomly distributed across the available shard replicas.
+
+However, you can specify a preference to influence the replica selection. The preference can be set to one of the following values:
+
+- `_local`: The operation will try to execute on a locally allocated shard replica, if possible. This can improve performance by reducing network overhead.
```

Comment:
```suggestion
- `_local`: The operation tries to execute on a locally allocated shard replica, if possible. This can improve performance by reducing network overhead.
```